hoot-core/src/main/cpp/hoot/core/filters/WaterwayCriterion.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * This file is part of Hootenanny.
+ *
+ * Hootenanny is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * --------------------------------------------------------------------
+ *
+ * The following copyright notices are generated automatically. If you
+ * have a new notice to add, please use the format:
+ * " * @copyright Copyright ..."
+ * This will properly maintain the copyright information. DigitalGlobe
+ * copyrights will be updated automatically.
+ *
+ * @copyright Copyright (C) 2015 DigitalGlobe (http://www.digitalglobe.com/)
+ */
+#ifndef WATERWAYCRITERION_H
+#define WATERWAYCRITERION_H
+
+#include "TagContainsFilter.h"
+
+namespace hoot
+{
+
+/**
+ * A filter that will either keep or remove waterways.
+ */
+class WaterwayCriterion : public TagContainsFilter
+{
+public:
+  WaterwayCriterion(FilterType type);
```

Comment:
My apologies for the mix of filters and criterions. I used filters initially and then kept confusing myself with the filter terminology. I have since moved to criterions and find I get confused a lot less. I'd like to replace all the filter code with criterions at some point...